toolchain/check/context.cpp

```diff
@@ -395,18 +395,18 @@ static auto GetCorePackage(Context& context, SemIRLoc loc)
   auto core_inst_id = context.LookupNameInExactScope(
       loc, core_name_id, SemIR::NameScopeId::Package,
       context.name_scopes().Get(SemIR::NameScopeId::Package));
-  if (!core_inst_id.is_valid()) {
-    context.DiagnoseNameNotFound(loc, core_name_id);
-    return SemIR::NameScopeId::Invalid;
+  if (core_inst_id.is_valid()) {
+    // We expect it to be a namespace.
+    if (auto namespace_inst =
+            context.insts().TryGetAs<SemIR::Namespace>(core_inst_id)) {
+      return namespace_inst->name_scope_id;
+    }
   }
 
-  // We expect it to be a namespace.
-  if (auto namespace_inst =
-          context.insts().TryGetAs<SemIR::Namespace>(core_inst_id)) {
-    return namespace_inst->name_scope_id;
-  }
-  // TODO: This should really diagnose the name issue.
-  context.DiagnoseNameNotFound(loc, core_name_id);
+  CARBON_DIAGNOSTIC(CoreNotFound, Error,
+                    "Package `Core` implicitly referenced here but not found.",
```

Comment:
These aren't independent clauses -- the "not found" is implicitly referring to the subject "package `Core`" -- so I think the "but" should not be preceded by a comma.

Reply:
I think there's a phrasing problem here; I have trouble reading it. Can I suggest instead something like:

Could not resolve implicit reference to package `Core`.

Reply:
Or just drop "here"

Reply:
Hm, I think I'm starting to see the (or at least an) issue: the "but" can be (mis)read as binding tightly to the "here" (as it would in "implicitly referenced here but not there") rather than to the "implicitly referenced" (as it would in "implicitly referenced but not found"). Your original suggestion of adding a comma seems like a reasonable way to fix that. While I think it's not strictly grammatically correct, our diagnostic messages are not written in formal English anyway. And looking through Clang's diagnostics, it uses "blah here, but [...]" with a dependent clause after the comma in various places too.

(I'm a bit hesitant to adopt your suggested "Could not resolve ..." phrasing -- I think we should prefer diagnostics to state what's wrong with the code, not how we discovered the problem, so "[you] referenced a thing that doesn't exist" seems better to me than "[I] could not resolve a reference to the thing" to me.)